students/mitch334/lesson09/mailroom.py

```diff
@@ -0,0 +1,244 @@
+        report += '-'*66 + '\n'
+        for donor in donors_sorted:
+            report += f'{donor.name:26} ${donor.donations_sum():>11.2f} {donor.donations_count():>11.0f}  ${donor.donations_avg():>12.2f}\n'
+
+        print(report)
+        return report
+
+
+    def send_letters(self):
+        """Create letters for each of the donors last donation"""
+
+        now = datetime.datetime.now()
+        now = str(now)[:-7].replace(' ','_').replace(':','')
+
+        location = tempfile.gettempdir()
+        files = []
+
+        for donor in self.donors:
+            file = location + '\\' + donor.name.replace(' ','-') + '__' + str(now) + '.txt'
+            files.append(file)
+
+            with open(file, 'w') as letter:
+                letter.write(donor.create_email(donor.donations[-1]))
+
+        print(f'Letters created and located at: {location}')
+        return files
+
+
+""" Initial donor list"""
+donors = DonorCollection([
+    Donor("William Gates, III", [100.00, 50.00]),
+    Donor("Jeff Bezos", [1000.00, 10.00, 500.00]),
+    Donor("Mark Zuckerberg", [200.00, 20.00, 50.00]),
+    Donor("Warren Buffet", [600.00, 300.00]),
+    Donor("Paul Allen", [1000.00])
+    ])
+
+
+def donor_sort_key_donations(donor):
+    return sum(donor.donations)
+
+
+def send_thankyou():
+    """Capture a donation and send a thankyou note"""
+
+    donor_name = ''
+    while donor_name != 'exit':
+        donor_name = input('Thank-You Menu:\n'
+                            '\tOptions:\n'
+                            "\t\tEnter 'list' for donor list\n"
+                            "\t\tEnter 'exit' to return to the main menu\n"
+                            '\tEnter full name of donor: ')
+
+        if donor_name.lower() == 'list':
+            print(donors.donors_list('list'))
+        elif donor_name.lower() == 'db':
+            print(donors.donors_list('db'))
+        elif donor_name.lower() == 'exit':
+            break
+        else:
+            current_donor = donors.add_donor(donor_name.title())
+
+            amount = None
+            while amount == None:
+                amount_input = input('Enter donation amount: $')
+                try:
+                    amount = round(float(amount_input),2)
+                    if amount <= 0:
+                        amount = None
+                        print('Error. Enter a positive donation amount.')
+
+                except ValueError:
+                    print('Error. Enter a valid donation amount.')
+
+
+            current_donor.add_donation(amount)
+
+            print(current_donor.create_email(amount))
+
+
+def quit():
+    print('Enjoy! :)')
+
+
+def invalid_menu():
+    print('Invalid option. Select 1, 2, 3, or 4.')
+
+# send_thankyou()
+# donors.create_report()
+# donors.send_letters()
+
+if __name__ == '__main__':
+
+    option = None
+    main_menu = {'1': send_thankyou, '2': donors.create_report, '3': donors.send_letters, '4': quit }
+
+    while option != '4':
```

Comment:
this is better in a `while True` loop that exists via option 4 on its own since you have the function defined.